driver/ixsystems/iscsi.py

```diff
@@ -141,7 +146,49 @@ def remove_export(self, context, volume):
         """
         pass
 
+    def check_connection(self):
+        # connection safety check for #27
+        if ix_utils.parse_truenas_version(self.common._system_version())[1] in ('12.0', '13.0'):
+            LOG.debug("Tanable: %s" % str(self.common._tunable()))
+            tunable = self.common._tunable()
+            # default value from Truenas 12 kern.cam.ctl.max_ports 256, kern.cam.ctl.max_luns 1024
+            lunallowed = 256
+            for item in tunable:
+                if (item.get('enabled') and
+                        item.get('var') == 'kern.cam.ctl.max_luns' and
+                        str(item.get('value')).isnumeric()):
+                    lunallowed = min(lunallowed, int(item['value']))
+                if (item.get('enabled') and
+                        item.get('var') == 'kern.cam.ctl.max_ports' and
+                        str(item.get('value')).isnumeric()):
+                    lunallowed = min(lunallowed, int(item['value']))
+            LOG.debug("Tunable OS max_luns/max_ports: %s" % lunallowed)
+
```

Comment:
common._tunable() returns a list of dict [{'var':'kern.cam.ctl.max_luns','enabled':True,'value':'256'},{'var':'kern.cam.ctl.max_ports','enabled':True,'value':'1024'}]